src/huggingface_hub/hf_api.py

```diff
@@ -1269,6 +1252,92 @@ def update_repo_visibility(
         r.raise_for_status()
         return r.json()
 
+    def rename_repo(
+        self,
+        from_id: str,
+        to_id: str,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+    ):
+        """
+        Rename a repository from namespace1/repo_name1 to namespace2/repo_name2
+
+        Renaming can be used for three use cases
+        - Renaming a repository within same user.
+        - Renaming a repository within same organization.
+        - Move repository from user to an organization.
+
+        Renaming does not work for:
+        - Moving a repository from an organization to another user or organization.
+        - Moving a repository from user A to user B.
```

Comment:
Should we leave add some guidance for the case where a user wants to transfer a model from org2org or user2user?